TYPO3.Media/Classes/TYPO3/Media/Domain/Model/ThumbnailGenerator/AbstractThumbnailGenerator.php

```diff
@@ -0,0 +1,108 @@
+<?php
+namespace TYPO3\Media\Domain\Model\ThumbnailGenerator;
+
+/*                                                                        *
+ * This script belongs to the TYPO3 Flow package "TYPO3.Media".           *
+ *                                                                        *
+ * It is free software; you can redistribute it and/or modify it under    *
+ * the terms of the GNU General Public License, either version 3 of the   *
+ * License, or (at your option) any later version.                        *
+ *                                                                        *
+ * The TYPO3 project - inspiring people to share!                         *
+ *                                                                        */
+
+use Imagine\Image\ImagineInterface;
+use TYPO3\Flow\Annotations as Flow;
+use Doctrine\ORM\Mapping as ORM;
+use TYPO3\Flow\Resource\ResourceManager;
+use TYPO3\Media\Domain\Model\Thumbnail;
+use TYPO3\Media\Exception;
+
+/**
+ * Abstract Thumbnail Generator
```

Comment:
Done done, we need a bit documentation around this feature, I can try to write something during the week and push a Doc PR